src/network/MissionControlProtocol.cpp

```diff
@@ -216,17 +220,44 @@ MissionControlProtocol::MissionControlProtocol(SingleClientWSServer& server)
 
 	this->_streaming_running = true;
 	this->_streaming_thread = std::thread(&MissionControlProtocol::videoStreamTask, this);
-	// TODO: add new thread to handle resending motor power values
+	this->_joint_repeat_running = true;
+	this->_joint_repeat_thread =
+		std::thread(&MissionControlProtocol::jointPowerRepeatTask, this);
 	// TODO: add new thread to periodically send motor status reports
 }
 
 MissionControlProtocol::~MissionControlProtocol() {
+	this->_joint_repeat_running = false;
+	this->_joint_repeat_thread.detach();
```

Comment:
In the destructor we should probably do more cleanup. For example, waiting for the joint thread to stop and then sending 0 power to all motors. We probably don't want to leave the motors running and rely on the watchdog to turn them off.

On that note, we should also definitely have a disconnection handler that shuts off all the motors. It would be pretty bad for MC to disconnect and leave all the motors running. Maybe the disconnection handler can be called from the destructor, that would fix both things.

Reply:
Yeah, that's a good idea. I'll get that implemented